jetty-server/src/main/java/org/eclipse/jetty/server/CookieCutter.java

```diff
@@ -390,28 +413,43 @@ else if (tokenstart >= 0)
 
     protected boolean isRFC6265RejectedCharacter(boolean inQuoted, char c)
     {
-        if (inQuoted)
+        // LEGACY test
+        if (_compliance == CookieCompliance.RFC6265_LEGACY)
         {
-            // We only reject if a Control Character is encountered
-            if (Character.isISOControl(c))
+            if (inQuoted)
             {
-                return true;
+                // We only reject if a Control Character is encountered
+                if (Character.isISOControl(c))
+                    return true;
             }
-        }
-        else
-        {
-            /* From RFC6265 - Section 4.1.1 - Syntax
-             *  cookie-octet  = %x21 / %x23-2B / %x2D-3A / %x3C-5B / %x5D-7E
-             *                  ; US-ASCII characters excluding CTLs,
-             *                  ; whitespace DQUOTE, comma, semicolon,
-             *                  ; and backslash
-             */
-            return Character.isISOControl(c) || // control characters
-                c > 127 || // 8-bit characters
-                c == ',' || // comma
-                c == ';'; // semicolon
+            else
+            {
+                /* From RFC6265 - Section 4.1.1 - Syntax
+                 *  cookie-octet  = %x21 / %x23-2B / %x2D-3A / %x3C-5B / %x5D-7E
+                 *                  ; US-ASCII characters excluding CTLs,
+                 *                  ; whitespace DQUOTE, comma, semicolon,
+                 *                  ; and backslash
+                 */
```

Comment:
I'd remove the comment because it's not what the code does (the code does not check for ` `, `"` and `\`.
Alternatively, can this block be merged with the block below which is identical and has a clarifying comment?

Reply:
I'll remove the comment.
It is not identical to the code below.  This is the original code and the code below is updated by @lorban.